community/index/src/main/java/org/neo4j/index/internal/gbptree/DynamicSizeUtil.java

```diff
@@ -28,28 +28,73 @@
  * Gather utility methods for reading and writing individual dynamic sized
  * keys. It thus define the layout for:
  * - Key pointer in offset array (K*), 2B
- * - Key size, 2B
- * - Value size, 2B
- * - Key tombstone, first bit in key size
+ * - keyValueSize, 1B-4B
+ * - Key tombstone, first bit in keyValueSize
+ *
+ * Format of key/value size is dynamic in itself, first byte being:
+ * <pre>
+ * [T,K,V,k,k,k,k,k]
+ * </pre>
+ * If {@code T} is set key is dead.
+ * If {@code K} is set the next byte contains the higher order bits of the key size.
+ * If {@code V} is set there is a value size to be read directly after key size.
+ * This first byte can fit key size <= 32 and we only need the second byte if key size is larger.
```

Comment:
I believe it should be `key size < 32`, since there's only 5 bits to store it on in the first byte.